File: ClinicControlCenter/Domain/Filters/UserFilter.cs
using System.Collections.Generic;
using SDK.Pagination;

namespace ClinicControlCenter.Domain.Filters
{
    public class UserFilter : PaginationFilter
    {
       public string Email { get; set; }

       public string Name { get; set; }

       public string Address { get; set; }

       public IEnumerable<string> UserTypes { get; set; } = new List<string>();
    }
}

File: ClinicControlCenter/ClientApp/src/app/system-management/models/UserFilter.ts
import { PaginationFilter } from "src/libs/util/base-api/src/lib/models/PaginationFilter";

export class UserFilter extends PaginationFilter {
  Email: string;
  Name: string;
  Address: string;
  UserTypes: string[];

  constructor(obj?: Partial<UserFilter>) {
    super();

    if (obj)
      for (const prop of Object.keys(obj)) {
        this[prop] = obj[prop];
      }
  }
}
